src/core/qgssettings.cpp

```diff
@@ -0,0 +1,259 @@
+  init( );
+}
+
+QgsSettings::QgsSettings( QSettings::Scope scope, const QString &organization,
+                          const QString &application, QObject *parent )
+{
+  mUserSettings = new QSettings( scope, organization, application, parent );
+  init( );
+}
+
+QgsSettings::QgsSettings( QSettings::Format format, QSettings::Scope scope,
+                          const QString &organization, const QString &application, QObject *parent )
+{
+  mUserSettings = new QSettings( format, scope, organization, application, parent );
+  init( );
+}
+
+QgsSettings::QgsSettings( const QString &fileName, QSettings::Format format, QObject *parent )
+{
+  mUserSettings = new QSettings( fileName, format, parent );
+  init( );
+}
+
+QgsSettings::QgsSettings( QObject *parent )
+{
+  mUserSettings = new QSettings( parent );
+  init( );
+}
+
+QgsSettings::~QgsSettings()
+{
+  delete mUserSettings;
+  delete mGlobalSettings;
+}
+
+
+void QgsSettings::beginGroup( const QString &prefix )
+{
+  mUserSettings->beginGroup( prefix );
+  if ( mGlobalSettings )
+  {
+    mGlobalSettings->beginGroup( prefix );
+  }
+}
+
+void QgsSettings::endGroup()
+{
+  mUserSettings->endGroup( );
+  if ( mGlobalSettings )
+  {
+    mGlobalSettings->endGroup( );
+  }
+}
+
+
+QStringList QgsSettings::allKeys() const
+{
+  QStringList keys = mUserSettings->allKeys( );
+  if ( mGlobalSettings )
+  {
+  for ( auto &s : mGlobalSettings->allKeys() )
+    {
+      if ( ! keys.contains( s ) )
+      {
+        keys.append( s );
+      }
+    }
+  }
+  return keys;
+}
+
+
+QStringList QgsSettings::childKeys() const
+{
+  QStringList keys = mUserSettings->childKeys( );
+  if ( mGlobalSettings )
+  {
+  for ( auto &s : mGlobalSettings->childKeys() )
+    {
+      if ( ! keys.contains( s ) )
+      {
+        keys.append( s );
+      }
+    }
+  }
+  return keys;
+}
+
+QStringList QgsSettings::childGroups() const
+{
+  QStringList keys = mUserSettings->childGroups( );
+  if ( mGlobalSettings )
+  {
+  for ( auto &s : mGlobalSettings->childGroups() )
+    {
+      if ( ! keys.contains( s ) )
+      {
+        keys.append( s );
+      }
+    }
+  }
+  return keys;
+}
+
+QVariant QgsSettings::value( const QString &key, const QVariant &defaultValue, const QgsSettings::Section section ) const
+{
+  QString pKey = prefixedKey( key, section );
+  if ( ! mUserSettings->value( pKey ).isNull() )
+  {
+    return mUserSettings->value( pKey );
+  }
+  if ( mGlobalSettings )
+  {
+    return mGlobalSettings->value( pKey, defaultValue );
+  }
+  return defaultValue;
+}
+
+bool QgsSettings::contains( const QString &key ) const
+{
+  return mUserSettings->contains( key ) ||
+         ( mGlobalSettings && mGlobalSettings->contains( key ) );
+}
+
+QString QgsSettings::fileName() const
+{
+  return mUserSettings->fileName( );
+}
+
+void QgsSettings::sync()
+{
+  return mUserSettings->sync( );
+}
+
+void QgsSettings::remove( const QString &key )
+{
+  mGlobalSettings->remove( key );
```

Comment:
@nirvn @NathanW2 the SEGFAULT error is here: is should be `mUserSettings` instead. I'm on holiday with limited connection and without my dev PC, I can fix it later this week.